themes/V3/5ePHB/snippets/tableOfContents.gen.js

```diff
@@ -1,41 +1,65 @@
 const _ = require('lodash');
 const dedent = require('dedent-tabs').default;
 
-const getTOC = (pages)=>{
+const getTOC = ()=>{
 
-	const recursiveAdd = (title, page, targetDepth, child, curDepth=0)=>{
+	const iframe = document.getElementById('BrewRenderer');
+	const iframeDocument = iframe.contentDocument || iframe.contentWindow.document;
+	const headings = iframeDocument.querySelectorAll('h1, h2, h3, h4, h5, h6');
+	const headerDepth = ['H1', 'H2', 'H3', 'H4', 'H5', 'H6'];
+
+	const res = [];
+
+	const pageMap = [];
+
+	const walkPages = ()=>{
+		let current = 0;
+		let skip = 0;
+		let reset = 0;
+		const pages = iframeDocument.querySelectorAll('.page');
+		_.each(pages, (page)=>{
+			current++;
+			if(page.querySelectorAll('.skipCounting').length > 0) {
```

Comment:
Can probably just use `querySelector` for locating the CSS classes. Should be simpler and faster because we only need to know if one exists.